### PR TITLE
fix(trading): quantum formatting changes

### LINKED ISSUE #2341: Use quantum formatting for liquidity provisions table
## Story

**As a** user
**I want** to see values in the liquidity provisions table that are more meaningful
**So that** I can better understand the real value of the numbers in the LP table

## Acceptance Criteria

- [ ] Numeric values in the LP table are formatted using asset.quantum
- [ ] I can interact with a column header to change the formatting to show the full raw number for that column
- [ ] There is a way to see the full number if I want to know the fully accurate number (e.g. with a tooltip)
- [ ] There is a way to find a definition of quantum, how it is calculated and why we use it in Console (i.e. [link to docs](https://github.com/vegaprotocol/documentation/issues/640))

## Tasks

- [ ] UX (if needed)
- [ ] Design (if needed)
- [ ] Team and stakeholder review
- [ ] Specs reviewed and created or adjusted
- [ ] Implementation
- [ ] Testing (unit and/or e2e)
- [ ] Code review
- [ ] QA review

## Additional details / background info

Created from: https://github.com/vegaprotocol/frontend-monorepo/issues/2132

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #2341 

# Description ℹ️

- [x] Use quantum formatting in liquidity provision tables.
- [x] Fixes quantum calculation by formatting settlement asset quantum with asset decimals
- [x] Fixes navigation to liquidity view
- [x] Fixes `Notional` and `Liquidation` estimates to be formatted with market decimals and shown in quote, not in settlement asset symbol. This is how it has been before but it was changed recently

# Demo 📺
image

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: libs/liquidity/src/lib/liquidity-table.tsx [hunk 1]
```diff
@@ -35,21 +37,152 @@ export interface LiquidityTableProps
...
+          minWidth: 160,
+          valueFormatter: assetDecimalsQuantumFormatter,
+          tooltipValueGetter: assetDecimalsFormatter,
+        },
+        {
+          headerName: t('Obligation'),
+          field: 'commitmentAmount',
+          type: 'rightAligned',
+          headerTooltip: t(
+            `The liquidity provider's obligation to the market, calculated as the liquidity commitment amount multiplied by the value of the stake_to_ccy_siskas network parameter to convert into units of liquidity volume. The obligation can be met by a combination of LP orders and limit orders on the order book.`
```
Comment (candida-d) [inline libs/liquidity/src/lib/liquidity-table.tsx:139]: I believe it's no longer called stake_to_ccy_siskas.

File: apps/trading-e2e/src/integration/market-liquidity.cy.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ describe('liquidity table - trading', { tags: '@smoke' }, () => {
     cy.get(rowSelector)
       .first()
       .find(colAverageEntryValuation)
-      .should('have.text', '685,852.93692');
+      .should('have.text', '685,852.94');
```
Comment (mattrussell36) [inline apps/trading-e2e/src/integration/market-liquidity.cy.ts:87]: chore (blocking): Can we add a new AC for this https://github.com/vegaprotocol/specs/blob/master/user-interface/5002-LIQP-liquidity.md

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading-e2e/src/integration/market-liquidity.cy.ts (hunk 2 above)
Comment (MadalinaRaicu) [reply apps/trading-e2e/src/integration/market-liquidity.cy.ts:87]: https://github.com/vegaprotocol/specs/pull/1769

File: libs/liquidity/src/lib/liquidity-table.tsx (hunk 1 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-table.tsx:139]: Yes, it is stakeToCcyVolume in the code. I'm updating the tooltip

Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-table.tsx:139]:
```suggestion
            `The liquidity provider's obligation to the market, calculated as the liquidity commitment amount multiplied by the value of the stake_to_ccy_volume network parameter to convert into units of liquidity volume. The obligation can be met by a combination of LP orders and limit orders on the order book.`
```

Comment (MadalinaRaicu) [inline libs/liquidity/src/lib/liquidity-table.tsx]:
```suggestion
            `The liquidity provider's obligation to the market, calculated as the liquidity commitment amount multiplied by the value of the stake_to_ccy_volume network parameter to convert into units of liquidity volume. The obligation can be met by a combination of LP orders and limit orders on the order book.`
```

File: apps/trading/lib/hooks/use-market-click-handler.ts [hunk 3]
```diff
@@ -20,20 +20,8 @@ export const useMarketClickHandler = (replace = false) => {
...
-      const link = Links[Routes.LIQUIDITY](selectedId);
-      if (metaKey) {
-        window.open(`/#${link}`, '_blank');
-      } else if (selectedId !== marketId || !isLiquidityPage) {
-        navigate(link, { replace });
-      }
-    },
-    [navigate, marketId, replace, isLiquidityPage]
-  );
+export const useMarketLiquidityClickHandler = () => {
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-market-click-handler.ts:23]: Why this change here?

Comment (mattrussell36) [reply apps/trading/lib/hooks/use-market-click-handler.ts:23]: I think we probably only want _blank if metaKey is true

**Review by mattrussell36 — APPROVED** [review]

File: apps/trading/lib/hooks/use-market-click-handler.ts (hunk 3 above)
Comment (MadalinaRaicu) [reply apps/trading/lib/hooks/use-market-click-handler.ts:23]: updated to keep the metakey. 
the method was changed because it was not working correctly. navigation on LP view was broken
even in mainnet

File: libs/deal-ticket/src/components/deal-ticket/deal-ticket-fee-details.tsx [hunk 4]
```diff
@@ -267,19 +267,7 @@ export const DealTicketFeeDetails = ({
...
-        ? liquidationEstimateBestCase
-        : liquidationEstimateWorstCase
-      ).toString(),
-      (liquidationEstimateBestCase > liquidationEstimateWorstCase
-        ? liquidationEstimateBestCase
-        : liquidationEstimateWorstCase
-      ).toString(),
-      assetDecimals,
-      quantum
+      marketDecimals
```
Comment (bglownia) [inline libs/deal-ticket/src/components/deal-ticket/deal-ticket-fee-details.tsx:270]: liquidation price should use assetDecimals, when using market decimals we are getting wrong values
image

Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/deal-ticket/deal-ticket-fee-details.tsx:270]: this might be a problem. is it a bug?
as far as I know from @barnabee `liquidation price` and `notional` are dependent on quote, and not settlement asset. so we should use market dp instead. 

why liquidation price uses asset dp instead?

Comment (bglownia) [reply libs/deal-ticket/src/components/deal-ticket/deal-ticket-fee-details.tsx:270]: It uses asset decimals and it has fractional part also.
image

@witgaw @barnabee should we change it in API to use market decimal places?

**Comment by JonRay15** [comment]
The value is coming in `asset dp` and this will not change imminently. 

We need to convert it using `asset dp` and then display it to the level of precision given in `market dp`.
